packages/reporter/src/commands/command.spec.tsx

```diff
@@ -295,6 +297,12 @@ describe('<Command />', () => {
 
       expect(component.find('.command-number')).to.have.text('1')
     })
+
+    it('displays the timeout progress indicator', () => {
+      const component = shallow(<Command model={model({ state: 'pending' })} aliasesWithDuplicates={null} />)
+
+      expect(component.find(Progress).first()).to.exist
```

Comment:
I do not believe this spec can fail because the `Progress` component is not being rendered conditionally... however, I'm not a React developer, so correct me if I'm wrong.

Do you understand why this spec was failing when you were looking for the DOM structure of `Progress` to assert the indicator was working?

Reply:
If we're going to use shallow mount, I would break this up into two specs:
1. Progress being rendered and passed the correct model
2. Progress having its own unit tests that are based on its internal structure + class

Alternatively, do a full mount (not shallow) and combine concerns 1 & 2

Reply:
A word of warning: testing timers can be particularly annoying at the unit/component level, so you could _instead_ choose to do an e2e spec to validate this, which may be easier to build and maintain 🙇‍♀️ 

Given the bugs we've been able to catch by manually testing, we should definitely add additional test coverage.

Reply:
That is correct, but I'm not quite sure how to test it otherwise. Because of the way that the animation progress and time is calculated, its nearly impossible to mock that behavior unless somehow `Date.now()` can be stubbed. So, I've reverted to just ensuring that it gets rendered.

I believe it was failing due to the use of `shallow` which prevents assertions on child components. However, all of the "logic" around displaying the bar has now been moved to CSS, so again it's not conditional that the `div` gets rendered, so I figured just checking the component would be sufficient.

Ideally, there would be a way to ensure that the animations were being calculated correctly because that is the important thing to test here, but that requires stubbing `Date.now()`.

Reply:
I agree that perhaps an `e2e` test would be better, also to ensure that the `log` model remains correct, however I am still unsure as to how to validate that the correct timing is calculated.